sklearn/calibration.py

```diff
@@ -25,6 +27,337 @@
+            The predicted class
+        """
+        X = check_array(X)
+        check_is_fitted(self, ["label_encoder_", "decision_threshold_"])
+
+        y_score = _get_binary_score(self.base_estimator, X, self.scoring,
+                                    self.pos_label)
+        return self.label_encoder_.inverse_transform(
+            (y_score > self.decision_threshold_).astype(int)
+        )
+
+
+class _CutoffClassifier(object):
+    """Cutoff point selection.
+
+    It assumes that base_estimator has already been fit, and uses the input set
+    of the fit function to select a cutoff point. Note that this class should
+    not be used as an estimator directly. Use the CutoffClassifier with
+    cv="prefit" instead.
+
+    Parameters
+    ----------
+    base_estimator : obj
+        The classifier whose decision threshold will be adapted according to
+        the acquired cutoff point. The estimator must have a decision_function
+        or a predict_proba
+
+    method : 'roc' or 'f_beta' or 'max_tpr' or 'max_tnr'
+        The method to use for choosing the cutoff point
+
+    beta : float in [0, 1]
+        beta value to be used in case method == 'f_beta'
+
+    scoring : str or None, optional (default=None)
+        The method to be used for acquiring the score. Can either be
+        "decision_function" or "predict_proba" or None. If None then
+        decision_function will be used first and if not available
+        predict_proba
+
+    pos_label : object
+        Label considered as positive during the roc_curve construction
+
+    threshold : float in [0, 1]
+        minimum required value for the true negative rate (specificity) in case
+        method 'max_tpr' is used or for the true positive rate (sensitivity) in
+        case method 'max_tnr' is used
+
+    Attributes
+    ----------
+    decision_threshold_ : float
+        Acquired decision threshold for the positive class
+    """
+    def __init__(self, base_estimator, method, beta, scoring, pos_label,
+                 threshold):
+        self.base_estimator = base_estimator
+        self.method = method
+        self.beta = beta
+        self.scoring = scoring
+        self.pos_label = pos_label
+        self.threshold = threshold
+
+    def fit(self, X, y):
+        """Select a decision threshold for the fitted model's positive class
+        using one of the available methods
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+
+        y : array-like, shape (n_samples,)
+            Target values
+
+        Returns
+        -------
+        self : object
+            Instance of self
+        """
+        y_score = _get_binary_score(self.base_estimator, X, self.scoring,
+                                    self.pos_label)
+        if self.method == 'f_beta':
+            precision, recall, thresholds = precision_recall_curve(
+                y, y_score, self.pos_label
```

Comment:
please pass pos_label as a named argument. we tend to avoid positional arguments except for things like X, y